### PR TITLE
Import more utilities from opm-core

### PR DESCRIPTION
**Pull request by akva2**
Plus a forgotten test.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this with downstreams opm-core=1209 please

**Comment by akva2** [comment]
hold this please, i need to amend with another test.

File: opm/core/linalg/blas_lapack.h [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
```
Comment (andlaus) [inline opm/core/linalg/blas_lapack.h]: just wondering: what's still using BLAS/LAPACK?

Comment (akva2) [reply opm/core/linalg/blas_lapack.h]: at the very least the tpfa code.

File: opm/core/utility/UniformTableLinear.hpp [hunk 2]
```diff
@@ -0,0 +1,268 @@
+/*
```
Comment (andlaus) [inline opm/core/utility/UniformTableLinear.hpp]: this file seems to provide largely the same functionality as opm-material's `Tabulated1DFunction`. as far as I can see it is not used by any code that leads to non-legacy `flow`. It's probably a good idea to move the interplation classes (`Tabulated1DFunction`, `UniformTabulated2DFunction`, `UniformXTabulated2DFunction` and `Spline` and their associated tests from opm-material to here, though.

File: opm/core/utility/buildUniformMonotoneTable.hpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
+/*
```
Comment (andlaus) [inline opm/core/utility/buildUniformMonotoneTable.hpp]: same as above: not used by non-legacy flow?

File: opm/core/utility/UniformTableLinear.hpp (hunk 2 above)
Comment (andlaus) [reply opm/core/utility/UniformTableLinear.hpp]: possibly, these classes should also get better names. any ideas?

File: opm/core/linalg/blas_lapack.h (hunk 1 above)
Comment (andlaus) [reply opm/core/linalg/blas_lapack.h]: Ok. can this be used by non-legacy flow? (I'd like to speed the build up for this one as much as possible...)

File: opm/core/utility/UniformTableLinear.hpp (hunk 2 above)
Comment (atgeirr) [reply opm/core/utility/UniformTableLinear.hpp]: I agree these things should be unified and put in opm-common. This PR just moves stuff, though! If you could make a list of files (remember any unit tests) that should be moved from opm-material we can do that, too. Then when the dust settles (== opm-core is dead, expected later this week), we'll start the refactoring/renaming job.

Comment (akva2) [reply opm/core/utility/UniformTableLinear.hpp]: first of all i don't think you are considering opm-upscaling. some of these things go into upscaling and simulators, so we cannot move them to simulators. the rest need a new home so we can kill core, and logic dictates they should be kept along the other related classes. we have a deliverable to focus on, then we will work on cleanup, removal and unification afterwards. stone by stone. thanks for pointing out things that can be cleaned up later, it will be done.

**Comment by akva2** [comment]
jenkins build this with downstreams opm-core=1209 please

File: opm/core/utility/UniformTableLinear.hpp (hunk 2 above)
Comment (andlaus) [reply opm/core/utility/UniformTableLinear.hpp]: okay then, let's proceed as proposed. Rebuilding flow with all dependencies currently takes an estimated factor 3 to 5 too long, though...

Comment (atgeirr) [reply opm/core/utility/UniformTableLinear.hpp]: > Rebuilding flow with all dependencies currently takes an estimated factor 3 to 5 too long, though...

Now where did you find that number... Recall that flow is not the only thing we build!

Comment (andlaus) [reply opm/core/utility/UniformTableLinear.hpp]: just my gutfeeling: Disregarding the well model and opm-parser you only need to compile the `flow_*.cc` compile units  for flow. on my system that takes an estimated 5th of the total time.

Comment (andlaus) [reply opm/core/utility/UniformTableLinear.hpp]: > Recall that flow is not the only thing we build!

True. That said, most of the time it is the only thing that we really *care* about!?

Comment (alfbr) [reply opm/core/utility/UniformTableLinear.hpp]: We care about the upscaling codes to. Actually, without them there would be no OPM.

Comment (andlaus) [reply opm/core/utility/UniformTableLinear.hpp]: yes, opm-upscaling is important, but IMO it should not slow down the build in 80 to 90% of the cases where it does not matter!

Comment (alfbr) [reply opm/core/utility/UniformTableLinear.hpp]: Yes, but there are more ways to get there. Not needing to rebuild opm-common as frequently would help. Allowing for turning off building of parts of the code is another. Not sure what you are aiming at though, e.g., opm-parser is an important part of flow, so your 80-90% number makes no sense in this context. Building opm-parser actually slows down building of opm-upscaling significantly, not flow.

**Comment by akva2** [comment]
all green, ready for merge.

File: opm/core/utility/UniformTableLinear.hpp (hunk 2 above)
Comment (andlaus) [reply opm/core/utility/UniformTableLinear.hpp]: okay, I've got some more resilient numbers: including everything, a complete optimized recompile without libecl up to opm-simulators takes 7 minutes and 25 seconds on my hyperthreaded 6-core machine while the stuff that strictly makes up `flow` only needs 1 minute and 10 seconds. of the total time, 50 seconds are spend on opm-parser and 57 seconds on ewoms. at least the ewoms stuff is not strictly necessary.

tl;dr: on my machine only about 16% of the build time are spend on building stuff directly related to the `flow` simulator.
